### PR TITLE
Milestone 2: Alter/Select Animals table

### PR DESCRIPTION
**Pull request by mirwaisfarahi**
Changes I have made in this PR:

- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Update all animals' weights that are negative to be their weight multiplied by -1.
- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = 't';
 SELECT * FROM animals WHERE name NOT LIKE '%Gabumon';
```
Comment (t-yanick) [inline queries.sql:9]: - [OPTIONAL] As it is a [requirement ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md)for this project, " `Add a column species of type string to your animals table. Modify your schema.sql file.` ", kindly add a species column to the animal table.

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) FROM animals WHERE (date_of_birth >= '01/01/1990' AND date_of_birth <= '01/01/2000') GROUP BY species;
```
Comment (t-yanick) [inline queries.sql:42]: - [OPTIONAL] As per project requirement, I think the date of birth should include all of the year 2000, something like so: 
 WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31', please kindly fix that.

**Review by t-yanick — APPROVED** [review]
Hi @Mirwais-Farahi, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (mirwaisfarahi) [reply queries.sql:9]: Thank you for the feedback @t-yanick , It has been fixed.

File: queries.sql (hunk 2 above)
Comment (mirwaisfarahi) [reply queries.sql:42]: Thank you for the feedback @t-yanick , It has been fixed.

**Pull request merged**